tests/components/vacuum/test_xiaomi.py

```diff
@@ -0,0 +1,161 @@
+"""The tests for the Xiaomi vacuum platform."""
+import asyncio
+from datetime import timedelta
+from unittest import mock
+
+import pytest
+
+from homeassistant.components.vacuum import (
+    ATTR_BATTERY_ICON,
+    ATTR_FAN_SPEED, ATTR_FAN_SPEED_LIST, DOMAIN,
+    SERVICE_LOCATE, SERVICE_RETURN_TO_BASE, SERVICE_SEND_COMMAND,
+    SERVICE_SET_FAN_SPEED, SERVICE_STOP,
+    SERVICE_TOGGLE, SERVICE_TURN_OFF, SERVICE_TURN_ON)
+from homeassistant.components.vacuum.xiaomi import (
+    ATTR_CLEANED_AREA, ATTR_CLEANING_TIME, ATTR_DO_NOT_DISTURB, ATTR_ERROR,
+    CONF_HOST, CONF_NAME, CONF_TOKEN, PLATFORM,
+    SERVICE_MOVE_REMOTE_CONTROL, SERVICE_MOVE_REMOTE_CONTROL_STEP,
+    SERVICE_START_REMOTE_CONTROL, SERVICE_STOP_REMOTE_CONTROL)
+from homeassistant.const import (
+    ATTR_SUPPORTED_FEATURES, CONF_PLATFORM, STATE_OFF, STATE_ON)
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture
+def mock_mirobo():
+    """Mock mock_mirobo."""
+    mock_vacuum = mock.MagicMock()
+    type(mock_vacuum.Vacuum().status()).data = {'test': 'raw'}
```

Comment:
I see... sorry, I'll change it.

I think in my first attempts was trying `mock_vacuum.Vacuum.status.XXX`, which wasn't working, and don't know how I ended up there... (but it works! ¿?)

I needed to create a pseudo-state for the vacuum to test the entity state props, because I was getting errors with `MagicMock` objects when accessing parameters. I tried until I've got it working...